src/main/java/com/dsrm/dsrmbackend/services/impl/ReservationServiceImpl.java

```diff
@@ -31,4 +34,18 @@ public Reservation addReservation(ReservationRequestDTO reservationRequestDTO) {
 
     @Override
     public Page<Reservation> getReservations(Pageable pageable) {return reservationRepo.findAll(pageable);}
+
+    @Override
+    public Page<Reservation> getReservations(Pageable pageable, Long userId ){
+        Specification<Reservation> reservationSpecification = Specification.where(null);
+
+        if (userId !=null) {
+            Optional<User> user = userRepo.findById(userId);
+            if (user.isPresent())
```

Comment:
we cannot do that in that way. If we will send wrong user id then it will return all reservations. It should return none.
Instead we can do it that way
```
        if (userId !=null) {
                reservationSpecification = reservationSpecification.and(
                        (Specification<Reservation>) (root, query, criteriaBuilder) -> criteriaBuilder.equal(root.get("user").get("id"), userId));
        }
``` and userRepo is not necessery so we can remove dependency from here